Index: src/org/eclipse/nebula/widgets/pshelf/PShelf.java
===================================================================
RCS file: /cvsroot/technology/org.eclipse.swt.nebula/org.eclipse.nebula.widgets.pshelf/src/org/eclipse/nebula/widgets/pshelf/PShelf.java,v
retrieving revision 1.4
diff -u -r1.4 PShelf.java
--- src/org/eclipse/nebula/widgets/pshelf/PShelf.java	19 Oct 2007 17:47:39 -0000	1.4
+++ src/org/eclipse/nebula/widgets/pshelf/PShelf.java	12 Sep 2008 07:41:27 -0000
@@ -416,10 +416,11 @@
             redraw();
             update();
             //workaround for SWT bug 193357
-            if (SWT.getPlatform().equals("carbon"))
+            /*if (SWT.getPlatform().equals("carbon"))
             {
             	getDisplay().readAndDispatch();
-            }
+            }*/
+            while(getDisplay().readAndDispatch());
             percentOfWork += .02;
         } 
         
@@ -472,13 +473,14 @@
             }
             
             sizeClients();
-            redraw(getClientArea().x,getClientArea().y, getClientArea().width, getClientArea().height, false);
+            redraw();
             update();
             //workaround for SWT bug 193357
-            if (SWT.getPlatform().equals("carbon"))
+            /*if (SWT.getPlatform().equals("carbon"))
             {
             	getDisplay().readAndDispatch();
-            }
+            }*/
+            while(getDisplay().readAndDispatch());
             percentOfWork += .02;
         } 
         
